org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/processing/JobManager.java

```diff
@@ -476,6 +476,7 @@ protected IStatus run(IProgressMonitor monitor) {
 
 						// must check for new job inside this sync block to avoid timing hole
 						if ((job = currentJob()) == null) {
+							onJobQueueEmpty();
```

Comment:
the both "notifyIdle(System.currentTimeMillis() - idlingStart);" will eventual fail . currentTimeMillis() is not monotonic
https://stackoverflow.com/questions/2978598/will-system-currenttimemillis-always-return-a-value-previous-calls

Reply:
this code is executed i a "synchronized (this)" block - that is probably unwanted. don't know if it can even  deadlock

Reply:
that code id called within a try block  "} catch (InterruptedException e) { // background indexing was interrupted"
If interrupted onJobQueueEmpty() may be not called at all :-(

Reply:
I think the surrounding of this new interception point was there from the beginning, of course there could be potential improvements that can be done here, But shall we keep them for a different focused issue ?

back to the last comment about interruption scenario, if the processingThread get interrupted this method should not processed right since the method supposed be executed in the interrupted thread.